mirest/ah-haven-space-sprinters
# #164694230 Users should receive notifications

**What does this PR do?**

- Allows users to receive in-app notifications and email notifications
- enable users to opt in and out of these notifications

**Description of Task to be completed?**

- [ ] Authors should receive in-app notifications when articles they favourited have new interactions like comments
- [x] Authors should receive in-app notifications when users they follow publish an article
- [ ] Authors should receive email notifications when users they follow publish an artice
- [ ] Authors should receive email notifications when articles they favourited have new interactions like comments
- [ ] Authors should also be able to opt in and out of these notiofications

**How should this be manually tested?**

- Clone the repository create a virtualenv and install all requirements 
- Run the command python manage.py runserver
- create atleast two user accounts
- ensure one user follows the other 
- create an article and run the endpoints bellow as a following user
 `Get/api/notifications` to get all notifications
 `Get/api/notifications<pk>` to get one notification
 `PUT/api/notifications` to mark all notifications as read
 `PUT/api/notifications<pk>` to mark one notification as read

**What are the relevant pivotal tracker stories?**

#164694230
